lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -474,6 +472,120 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		return $user_cpt;
 	}
 
+	/**
+	 * Saves a new user variation into the database.
+	 *
+	 * @param array $args {
+	 *     Arguments to add a style variation.
+	 *
+	 *     @type string title Required. Global styles variation name.
+	 *     @type string global_styles Required. Global styles settings as a JSON string.
+	 *     @type string $stylesheet Required. Slug of the theme associated with these global styles.
+	 *     @type string $post_name Optional. Post name.
+	 * }
+	 *
+	 * @return int|WP_Error Post ID of the new variation or error if insertion failed.
+	 */
+	public static function add_user_global_styles_variation( $args ) {
+		$theme = wp_get_theme();
+
+		/*
+		 * Bail early if the theme does not support a theme.json.
+		 *
+		 * Since wp_theme_has_theme_json only supports the active
+		 * theme, the extra condition for whether $theme is the active theme is
+		 * present here.
+		 */
+		if ( $theme->get_stylesheet() === $args['stylesheet'] && ! wp_theme_has_theme_json() ) {
+			return new WP_Error( __( 'Theme does not have theme.json', 'gutenberg' ) );
+		}
+
+		$post_args = array(
+			'post_content' => $args['global_styles'],
+			'post_status'  => 'publish',
+			'post_title'   => $args['title'],
+			'post_type'    => 'wp_global_styles',
+			'tax_input'    => array(
+				'wp_theme' => array( $args['stylesheet'] ),
+			),
+		);
+
+		if ( ! empty( $args['post_name'] ) ) {
+			$post_args['post_name'] = $args['post_name'];
+		}
+
+		$post_id = wp_insert_post(
+			$post_args,
+			true
+		);
+
+		if ( empty( $args['post_name'] ) ) {
+			$post_id = wp_update_post(
+				array(
+					'ID'        => $post_id,
+					'post_name' => self::generate_user_style_variation_post_name( $post_id, $args['stylesheet'] ),
+				)
+			);
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Generate a post name for a variation with the given ID.
+	 *
+	 * @since 6.2
+	 *
+	 * @param int|null $variation_id Variation ID.
+	 * @param string   $stylesheet   Stylesheet.
+	 * @return string Post name.
+	 */
+	public static function generate_user_style_variation_post_name( $variation_id, $stylesheet ) {
+		return "wp-global-styles-{$stylesheet}-variation-{$variation_id}";
```

Comment:
When is `generate_user_style_variation_post_name` called with `$variation_id` as `null`?